File: sample/fftw/fftwsample.c
#include <stdio.h>
#include <fftw3.h>

int main() {
    int N = 16;
    double in[N];
    fftw_complex out[N/2 + 1];
    fftw_plan p;

    // 入力データを初期化
    for (int i = 0; i < N; i++) {
        in[i] = i + 1.0;
    }

    // FFTの計画を作成
    p = fftw_plan_dft_r2c_1d(N, in, out, FFTW_ESTIMATE);

    // FFTの実行
    fftw_execute(p);

    // 出力結果を表示
    printf("FFT output:\n");
    for (int i = 0; i < N/2 + 1; i++) {
        printf("out[%d] = %2.2f + %2.2fi\n", i, out[i][0], out[i][1]);
    }

    // メモリを解放
    fftw_destroy_plan(p);
    fftw_cleanup();

    return 0;
}

File: sample/fftw/fftwsample.f90
program fftw_example
  use fftw3
  implicit none

  integer, parameter :: N = 16
  real(8) :: in(N)
  complex(8) :: out(N/2 + 1)
  type(fftw_plan) :: plan
  integer :: i

  ! 入力データの初期化
  do i = 1, N
     in(i) = dble(i)
  end do

  ! FFTの計画を作成
  plan = fftw_plan_dft_r2c_1d(N, in, out, FFTW_ESTIMATE)

  ! FFTの実行
  call fftw_execute(plan)

  ! 結果を表示
  print *, "FFT output:"
  do i = 1, N/2 + 1
     print *, "out(", i, ") = ", out(i)
  end do

  ! メモリを解放
  call fftw_destroy_plan(plan)
  call fftw_cleanup()

end program fftw_example
